content/terraform-docs-common/docs/cloud-docs/registry/publish-modules.mdx

```diff
@@ -5,123 +5,198 @@ description: Use the HCP Terraform private registry to publish and share private
+
+For example, specifying `app` as a tag prefix for the registry module instructs HCP Terraform to select only Git tags with the prefix `app` in them. As a result, HCP Terrraform selects the module tagged with `appv1.0.0` and publishes `v1.0.0` to the registry. 
+
+Refer to [Enable publishing from a single repository](#enable-publishing-from-a-single-repository)
+
+### Module 
 
-Members of the [owners team](/terraform/cloud-docs/users-teams-organizations/permissions#organization-owners) and teams with [Manage Private Registry permissions](/terraform/cloud-docs/users-teams-organizations/permissions#manage-private-registry) can publish and delete modules from the private registry.
+By default, HCP Terraform and Terraform Enterprise require a separate repository for each module. As a result, each module must use the following three-part naming convention:
 
+`terraform-<PROVIDER>-<NAME>`
 
-## Preparing a Module Repository
+The `<NAME>` segment reflects the type of infrastructure the module manages and `<PROVIDER>` is the main provider where it creates the infrastructure. 
 
-After you configure at least one [connection to a VCS provider][vcs], you can publish a new module by specifying a properly formatted VCS repository (details below). The registry automatically detects the rest of the information it needs, including the module's name and its available versions.
+The `<PROVIDER>` segment must be all lowercase. The `<NAME>` segment can contain additional hyphens. 
 
-A module repository must meet all of the following requirements before you can add it to the registry:
+Examples: 
 
-- **Location and permissions:** The repository must be in one of
-  your configured [VCS providers][vcs], and HCP Terraform's VCS user account must have admin access to the repository. The registry needs admin access to create the webhooks to import new module versions. GitLab repositories must be in the main organization or group, and not in any subgroups.
+- `terraform-google-vault`
+- `terraform-aws-ec2-instance`
 
-- **Named `terraform-<PROVIDER>-<NAME>`:** Module repositories must use this
-  three-part name format, where `<NAME>` reflects the type of infrastructure
-  the module manages and `<PROVIDER>` is the main provider where it creates that
-  infrastructure. The `<PROVIDER>` segment must be all lowercase. The `<NAME>`
-  segment can contain additional hyphens. Examples: `terraform-google-vault` or
-  `terraform-aws-ec2-instance`.
+When the **Publish modules to your private registry from a single repository** option is enabled,you can store multiple modules in a single repository. As a result, modules do not need to follow the naming convention. Refer to the following topics for more information:
```

Comment:
```suggestion
When the **Publish modules to your private registry from a single repository** option is enabled, you can store multiple modules in a single repository. As a result, these repositories do not need to follow the naming convention. Refer to the following topics for more information:
```

I think we're referring to the naming of the VCS repository, not the module itself?